src/plugins/topic_viewer/TopicViewer.qml

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+import QtQml.Models 2.2
+import QtQuick 2.0
+import QtQuick.Controls 1.4
+import QtQuick.Controls 2.2
+import QtQuick.Controls.Styles 1.4
+import QtQuick.Controls.Material 2.1
+import QtQuick.Layouts 1.3
+
+TreeView {
+    id:tree
+    model: TopicsModel
+
+    Layout.minimumHeight: 400
+    Layout.minimumWidth: 300
+    anchors.fill: parent
+
+    property int itemHeight: 30;
+
+    // =========== Colors ===========
+    property color oddColor: (Material.theme == Material.Light) ?
+                              Material.color(Material.Grey, Material.Shade100):
+                              Material.color(Material.Grey, Material.Shade800);
+
+    property color evenColor: (Material.theme == Material.Light) ?
+                               Material.color(Material.Grey, Material.Shade200):
+                               Material.color(Material.Grey, Material.Shade900);
+
+    property color highlightColor: Material.accentColor;
+
+
+    verticalScrollBarPolicy: Qt.ScrollBarAsNeeded
+    horizontalScrollBarPolicy: Qt.ScrollBarAlwaysOff
+    headerVisible: false
+    headerDelegate: Rectangle {
+        visible: false
+    }
+    backgroundVisible: false;
+    TableViewColumn
+    {
+        role: "name";
+//        width: parent.width;
+    }
+
+    // =========== Selection ===========
+    selection: ItemSelectionModel {
+        model: tree.model
+    }
+    selectionMode: SelectionMode.SingleSelection
+
+    // =========== Delegates ============
+    rowDelegate: Rectangle
+    {
+        id: row
+        color: (styleData.selected)? highlightColor :
+                                     (styleData.row % 2 == 0) ? evenColor : oddColor
+        height: itemHeight;
+    }
+
+    itemDelegate: Item {
+        id: item
+
+        // for fixing the item position
+        // item pos changes randomly when drag happens (with the copy drag)
+        anchors.top: parent.top
+        anchors.right: parent.right
+
+        Drag.mimeData: { "text/plain" :  model.topic + "," + model.path }
```

Comment:
How about using the same pattern as Gazebo-classic? `topic?p=/fieldA/fieldB`

That works like a URI, so it should be easier to break apart using the `ignition::common::URI` class.

Reply:
Also, I get messages like this when expanding items:

`[GUI] [Wrn] [Application.cc:649] [QT] file::/TopicViewer/TopicViewer.qml:83: TypeError: Cannot read property 'topic' of null`

It would be good to check if `model` is `null`.